src/components/flats/FlatListItem.tsx

```diff
@@ -16,7 +16,7 @@ function FlatListItem(props: FlatListItemProps) {
     }
 
     const handleEditClick = () => {
-        navigate(`/flats/${props.flat.id}`);
+        navigate(`/flats/${props.flat.id}/edit`, {state: props.flat});
```

Comment:
Jak już zrobiłeś pobieranie edytowane flata z api to nie musisz tego state'a już przekazywać

Reply:
done